ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala

```diff
@@ -0,0 +1,145 @@
+        .setCommandId("commandId")
+        .setSubmitter("aSubmitter"))
+    .setSubmissionSeed(ByteString.copyFromUtf8("a" * 32))
+    .build
+  private val aTransactionEntrySummary = DamlTransactionEntrySummary(aDamlTransactionEntry)
+  private val aRecordTime = Timestamp(100)
+
+  "deduplicateCommand" should {
+    "continue if record time is not available" in {
+      val instance =
+        new TransactionCommitter(theDefaultConfig, mock[Engine], metrics, inStaticTimeMode = false)
+      val context = new FakeCommitContext(recordTime = None)
+
+      val actual = instance.deduplicateCommand(context, aTransactionEntrySummary)
+
+      actual match {
+        case StepContinue(_) => succeed
+        case StepStop(_) => fail
+      }
+    }
+
+    "continue if record time is after deduplication time in case a deduplication entry is found" in {
+      val instance =
+        new TransactionCommitter(theDefaultConfig, mock[Engine], metrics, inStaticTimeMode = false)
+      val dedupValue = DamlStateValue.newBuilder
+        .setCommandDedup(
+          DamlCommandDedupValue.newBuilder.setDeduplicatedUntil(buildTimestamp(aRecordTime)))
+        .build
+      val inputs =
+        Map(Conversions.commandDedupKey(aTransactionEntrySummary.submitterInfo) -> Some(dedupValue))
+      val context =
+        new FakeCommitContext(recordTime = Some(aRecordTime.addMicros(1)), inputs = inputs)
+
+      val actual = instance.deduplicateCommand(context, aTransactionEntrySummary)
+
+      actual match {
+        case StepContinue(_) => succeed
+        case StepStop(_) => fail
+      }
+    }
+
+    "produce rejection log entry in case record time is on or before deduplication time" in {
+      val instance =
+        new TransactionCommitter(theDefaultConfig, mock[Engine], metrics, inStaticTimeMode = false)
+      for ((recordTime, deduplicationTime) <- Iterable(
+          (aRecordTime, aRecordTime),
+          (aRecordTime, aRecordTime.addMicros(1)))) {
+        val dedupValue = DamlStateValue.newBuilder
+          .setCommandDedup(DamlCommandDedupValue.newBuilder.setDeduplicatedUntil(
+            buildTimestamp(deduplicationTime)))
+          .build
+        val inputs =
+          Map(
+            Conversions.commandDedupKey(aTransactionEntrySummary.submitterInfo) -> Some(dedupValue))
+        val context =
+          new FakeCommitContext(recordTime = Some(recordTime), inputs = inputs)
+
+        val actual = instance.deduplicateCommand(context, aTransactionEntrySummary)
+
+        actual match {
+          case StepContinue(_) => fail
+          case StepStop(actualLogEntry) =>
+            actualLogEntry.hasTransactionRejectionEntry shouldBe true
+        }
+      }
+    }
+  }
+
+  "validateLedgerTime" should {
+    "continue if record time is not available" in {
+      val instance =
+        new TransactionCommitter(theDefaultConfig, mock[Engine], metrics, inStaticTimeMode = false)
+      val context = new FakeCommitContext(recordTime = None)
```

Comment:
Could you also check that the same happens even if a time model is present as an input?

Reply:
What do you mean? There must always be a time model present otherwise we cannot resolve the current configuration (i.e., `Configuration.decode` will throw in case there's no time model present). Outside of pre-execution we will always check against the time-model whereas with pre-execution we will only perform time-bound checks during post-execution by a different component.

Reply:
Right, so this test is stricter than it should be, i.e. it should also provide for a configuration input because it would be valid for the tested code to fail without it. Alternatively, if this is the intended spec for the foreseeable future, I suggest changing the description into e.g. `continue without accessing the ledger configuration if record time is not available`.

Reply:
It's the simplest you can write to test this code path ; ) OK, changed the description.